### DISCUSSION TITLE
ScriptReference/TextureImporter.GetAutomaticFormat

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#229] [Notes]
# ScriptReference/TextureImporter.GetAutomaticFormat

https://docs.unity3d.com/ScriptReference/TextureImporter.GetAutomaticFormat.html

**Comment by mvi** [comment]
Note that this will return the wrong format if the platform has a format override set but override is disabled. Generally it looks like turning off the platform override in the UI will reset the platform format to Automatic, however our project has recently experienced lots of textures with override turned off but still retaining non-automatic overrides. It's not clear why this has happened (possibly a bug in Unity). This means that the Unity UI for example will tell us that a texture is DXT1 (which it is), but calling GetAutomaticFormat will say DXT5 because it had previously been overriden to DXT5.  
To work around this you can call `TextureImporter.DefaultFormatFromTextureParameters` via reflection and pass in the platform settings if override is enabled (otherwise pass in the default platform settings).
